test/unit/modules/api-translateSpec.js

```diff
@@ -35,13 +35,59 @@ describe('Unit: Testing api-translate', function () {
 
     describe('when translating with the service...', function () {
 
-        it('should return the translated string if key exists', inject(function ($translate) {
-            $translate('SALUTE').should.be.equal('Hello');
-        }));
+        describe('using asnyc API...', function () {
+
+            it('should return the translated string if key exists', inject(function ($q, $rootScope, $translate) {
+                var deferred = $q.defer(),
```

Comment:
Too complicated and using a workaround with $rootScope.$apply().
Mocha supports async tests: https://mochajs.org/#asynchronous-code
Please do not use $q nor $rootScope.

Reply:
This is how the `$translate` service is unit tested in `angular-translate` library (in fact, here we are not testing anything from Appverse):

https://github.com/angular-translate/angular-translate/blob/eda9f29b3343d84f43f9fa30a39a1ae636d1c494/test/unit/service/translate.spec.js#L1940-L1954

I tried the following and works using the `done()` function from Mocha, but the `$rootScope.$apply()` is necessary to trigger the `$translate` Promise handler:

```
           it('should return the translated string if key exists', function (done) {
               inject(function ($translate, $rootScope) {
                   $translate('SALUTE').then(function (translation) {
                       translation.should.be.equal('Hello');
                       done();
                    }, function () {
                        assert.fail('Could not translate existing key');
                    });
                    $rootScope.$apply();
               })
           });
```

If this is OK to you, I'll update the pull request.

Reply:
> This is how the `$translate` service is unit tested in `angular-translate` library (in fact, here we are not testing anything from Appverse):

Good point, we should test the translate directive instead, please replace tests with:
```js
var element = $compile("<p translate>SALUTE</p>")($rootScope);
$rootScope.$digest();
element.html().should.be.equal("Hello");
```
No need to have async and sync tests nor to test $translate service.

Reply:
> > This is how the `$translate` service is unit tested in `angular-translate` library (in fact, here we are not testing anything from Appverse):
> 
> Good point, we should test the translate directive instead, please replace tests with:
> 
> ```js
> var element = $compile("<p translate>SALUTE</p>")($rootScope);
> $rootScope.$digest();
> element.html().should.be.equal("Hello");
> ```

This is already tested:

https://github.com/Appverse/appverse-web-html5-core/blob/05e1a0fb8cb2a35606f331fbdbf0288bfd027c31/test/unit/modules/api-translateSpec.js#L61-L80

> No need to have async and sync tests nor to test $translate service.

So if I understood, can I just remove those two `$translate` service tests?

Reply:
You can remove them as we indeed only decorate the `translate` directive not the service. Test coverage should not decrease, it can be seen in `reports/coverage` folder.

Reply:
@aecz done.